packages/ag-charts-community/src/chart/series/chord/chordSeries.ts

```diff
@@ -351,25 +391,102 @@ export class ChordSeries extends DataModelSeries<
+                datumSelection: this.highlightNodeSelection,
+            });
+            await this.updateNodeNodes({ datumSelection: this.highlightNodeSelection, isHighlight: true });
+
+            this.highlightLinkSelection = await this.updateLinkSelection({
+                nodeData: [],
+                datumSelection: this.highlightLinkSelection,
+            });
+            await this.updateLinkNodes({ datumSelection: this.highlightLinkSelection, isHighlight: true });
+        } else if (highlightedDatum?.type === ChordDatumType.Link) {
+            this.focusNodeSelection = await this.updateNodeSelection({
+                nodeData: [highlightedDatum.fromNode, highlightedDatum.toNode],
+                datumSelection: this.focusNodeSelection,
+            });
+            await this.updateNodeNodes({ datumSelection: this.focusNodeSelection, isHighlight: false });
+
+            this.focusLinkSelection = await this.updateLinkSelection({
+                nodeData: [highlightedDatum],
+                datumSelection: this.focusLinkSelection,
+            });
+            await this.updateLinkNodes({ datumSelection: this.focusLinkSelection, isHighlight: false });
+
+            this.highlightNodeSelection = await this.updateNodeSelection({
+                nodeData: [],
+                datumSelection: this.highlightNodeSelection,
+            });
+            await this.updateNodeNodes({ datumSelection: this.highlightNodeSelection, isHighlight: true });
+
+            this.highlightLinkSelection = await this.updateLinkSelection({
+                nodeData: [highlightedDatum],
+                datumSelection: this.highlightLinkSelection,
+            });
+            await this.updateLinkNodes({ datumSelection: this.highlightLinkSelection, isHighlight: true });
+        } else {
+            this.focusNodeSelection = await this.updateNodeSelection({
+                nodeData: [],
+                datumSelection: this.focusNodeSelection,
+            });
+            await this.updateNodeNodes({ datumSelection: this.focusNodeSelection, isHighlight: false });
+
+            this.focusLinkSelection = await this.updateLinkSelection({
+                nodeData: [],
+                datumSelection: this.focusLinkSelection,
+            });
+            await this.updateLinkNodes({ datumSelection: this.focusLinkSelection, isHighlight: false });
+
+            this.highlightNodeSelection = await this.updateNodeSelection({
+                nodeData: [],
+                datumSelection: this.highlightNodeSelection,
+            });
+            await this.updateNodeNodes({ datumSelection: this.highlightNodeSelection, isHighlight: true });
+
+            this.highlightLinkSelection = await this.updateLinkSelection({
+                nodeData: [],
+                datumSelection: this.highlightLinkSelection,
+            });
+            await this.updateLinkNodes({ datumSelection: this.highlightLinkSelection, isHighlight: true });
+        }
```

Comment:
Any way we can improve the signal to noise ratio in terms of readability here, it looks like there are 4 `this.updateNodeSelection()` calls happening on all 3 branches, and only some of the parameters are changed?

Reply:
Updated